### PR TITLE
2.x: Add missing Resource Observer for Maybe, Completable & Single and adjust some Javadoc

### LINKED ISSUE #4517: 2.x: ResourceSubscriber & ResourceObserver
Why is `ResourceSubscriber` using `CompositeDisposable` and `ResourceObserver` using `ListCompositeDisposable`?

Also the java documentation does not seem to fit of the disposables.
`/** The resource composite, can be null. */`

It's initialized and made final and adding a `null` Disposable will throw a `NullPointerException`.

In addition is there a need for ResourceMaybeObserver and the alike implementation for Single & Completable?

### PR DESCRIPTION
**Pull request by vanniktech**
Fixes #4517

### CODE REVIEW

File: src/test/java/io/reactivex/observers/ResourceObserverTest.java [hunk 1]
```diff
@@ -62,10 +65,6 @@ public void onComplete() {
 
             dispose();
         }
-
```
Comment (akarnokd) [inline src/test/java/io/reactivex/observers/ResourceObserverTest.java:65]: This was for coverage on the protected cancel method.

**Comment by akarnokd** [comment]
It might be worth considering removing the protected `cancel()` methods as they are duplicates of `dispose`. For `ResourceSubscriber` I don't know because it works with `Subscription.cancel()` but also is a `Disposable` so `dispose()` is there as well.

**Comment by vanniktech** [comment]
So should I remove the `cancel()` method on all `Resource*Observer`?

**Comment by akarnokd** [comment]
@JakeWharton your opinion on the removal?

**Comment by JakeWharton** [comment]
I'm supportive of removing `cancel()`. I much prefer when there's only one way to do something!

**Comment by akarnokd** [comment]
:+1:

**Pull request merged**